File: introduccion orientada a objetos/ejer1.java
// Ejercicio 1: Clase Persona
class Persona {
    private String nombre;
    private int edad;
    private String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public void saludar() {
        System.out.println("Hola, soy " + nombre + " de " + ciudad);
    }

    public boolean esMayorDeEdad() {
        return edad >= 18;
    }

    public static void main(String[] args) {
        Persona p1 = new Persona("Juan", 20, "La Paz");
        Persona p2 = new Persona("Ana", 17, "Cochabamba");
        Persona p3 = new Persona("Luis", 25, "Santa Cruz");

        p1.saludar();
        System.out.println("Mayor de edad: " + p1.esMayorDeEdad());

        p2.saludar();
        System.out.println("Mayor de edad: " + p2.esMayorDeEdad());

        p3.saludar();
        System.out.println("Mayor de edad: " + p3.esMayorDeEdad());
    }
}

File: introduccion orientada a objetos/ejer1.py
class Persona:
    def __init__(self, nombre: str, edad: int, ciudad: str):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad
    
    def saludar(self):
        print(f"Hola, soy {self.nombre} de {self.ciudad}")
    
    def es_mayor_de_edad(self):
        return self.edad >= 18
    
    def main():
        p1 = Persona("Juan", 20, "La Paz")
        p2 = Persona("Ana", 17, "Cochabamba")
        p3 = Persona("Luis", 25, "Santa Cruz")
        
        p1.saludar()
        print(f"Mayor de edad: {p1.es_mayor_de_edad()}")
        
        p2.saludar()
        print(f"Mayor de edad: {p2.es_mayor_de_edad()}")
        
        p3.saludar()
        print(f"Mayor de edad: {p3.es_mayor_de_edad()}")

if __name__ == "__main__":
    Persona.main()
